### DISCUSSION TITLE
Small size after apply fitz.TOOLS.set_small_glyph_heights(True)

### DISCUSSION THREAD

**Discussion by Muhammadraafat1** [pymupdf/PyMuPDF#3514] [Q&A] [answered]
### Description of the bug

I was have a issue "After applying redactions with Page.add_redact_annot() some text disappears" but I applied set_small_glyph_heights and it works well, but the fontsize became small even I determined it by 20 and it's not effected

for example:
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/7d8b6453-b369-4e81-b79c-cda618265d76)

![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/c690181a-04a3-4b71-82a4-71195de39afc)

### How to reproduce the bug

``   
```
for page in doc:
        fitz.TOOLS.set_small_glyph_heights(True)
        for word , replace in `replacements.items():`
            wlist = page.search_for(word)
            for rect in wlist:
                redaction = page.add_redact_annot(
                    rect,
                    text=replace,
                    align=fitz.TEXT_ALIGN_CENTER,
                    `fontsize=20`
                )
        page.apply_redactions()
    doc.save(output_pdf)

```
``

### PyMuPDF version

1.24.4

### Operating system

Windows

### Python version

3.12

**Comment by JorjMcKie** [comment]
The fontsize will be automatically adjusted to smaller values until the full text string fits into the redaction (!) rectangle.
If you reduced the redaction rectangle height, then this is what happens.

**Comment by Muhammadraafat1** [comment]
@JorjMcKie 
There is no way to make customization to keep fontsize as the font of original text?

**Comment by JorjMcKie** [comment, marked as answer]
Have converted this to a Discussions post.

The automatic text replacement implemented inside redactions is a convenience - inevitably coming with some restrictions as to how much this can be influenced.

If your needs go beyond, you must adapt your approach and do text replacement in multiple steps:

1. Extract old text and determine its properties like font size, text color, maybe also font, insertion point etc.
1. Remove the old text **_without_** specifying a replacement.
2. Insert the new text afterwards using desired properties. This will then not be under control of the redaction logic. You can also choose any of the text insertion variants `insert_text`, `insert_textbox`, `fill_textbox`, `insert_htmlbox` and benefit from their flexibility.
